### ISSUE TITLE
Suitable model suite

### ISSUE THREAD

**Issue by MartinBernstorff** [Aarhus-Psychiatry-Research/psycop-t2d#43]
- [x] Logistic regression
- [ ] Naive bayes
- [ ] Explainable boosting machines: https://interpret.ml/docs/ebm.html

**Comment by martbern** [comment]
Problem: If we specify multiple models in the conf, Optuna might think that changing a hyperparam for one model is a new experiment for another model.

We probably want to use hydra composition to specify model and its hyperparams as a "chunk". Then spawn a set of Optuna experiments for each model

**Comment by KennethEnevoldsen** [comment]
## logistic regression grid:
- continuous_variables z-score normalization T/F (standardScaler)
alpha=1.0,  [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 0.0, 1.0, 10.0, 100.0]
l1_ratio=0.5  arange(0, 1, 0.01)
fit_intercept=True
precompute=False
max_iter=1000
copy_X=True
tol=0.0001
warm_start=False
positive=False
random_state=${project.seed}
selection='cyclic'

## Naive bayes
- continuous_variables z-score normalization T/F (standardScaler)
- GaussianNB()

## EBMs
Documentiation: https://interpret.ml/docs/ebm.html#api

- max_bins=256
max_interaction_bins=32
binning='quantile'
mains='all'
interactions=10 [0, 5, 10, 50]
outer_bags=8
inner_bags=0
learning_rate=0.01   [0.001, 0.01, 0.1]
validation_size=0.15.  [0.20, 0.15, 0.10, 0.05]
early_stopping_rounds=50 
early_stopping_tolerance=0.0001
max_rounds=5000
min_samples_leaf=2,  [1, 2]
max_leaves=3 [2, 3, 4, 8, 16]
n_jobs=- 2
random_state=${project.seed}

For this model we will also need to add visualization.

**Comment by MartinBernstorff** [comment]
@KennethEnevoldsen, jeg får den her warning fra "interpret": 
`[2022-07-15 11:46:35,949][interpret.utils.all][WARNING] - Passing a numpy array to schema autogen when it should be dataframe.`

Det lyder som noget der kan give problemer med interpretations. Vi kan godt lade modellen ligge i dvale indtil vi har lyst, hvis du har det bedst med det - så lukker du bare issuet 👍

**Comment by KennethEnevoldsen** [comment]
Is this only a warning? Why not simply pass a dataframe?

**Comment by MartinBernstorff** [comment]
Usikker på ovenstående spørgsmål, men tænker måske det er nemmere at du debugger den? :-)
